src/code/z_message_nes.c

```diff
@@ -288,8 +290,1589 @@ void Message_LoadOwlWarpTextNES(PlayState* play, s32* offset, f32* arg2, s16* de
+                    if (msgCtx->unk120D6) {
+                        msgCtx->unk120D4 += 25;
+                        if (msgCtx->unk120D4 >= 100) {
+                            msgCtx->unk120D6 = false;
+                        }
+                    } else {
+                        msgCtx->unk120D4 -= 25;
+                        if (msgCtx->unk120D4 <= -100) {
+                            msgCtx->unk120D6 = true;
+                        }
+                    }
+
+                    if (D_801D07DC[(s16)(character - 1)][0] + msgCtx->unk120D4 < 0) {
+                        msgCtx->textColorR = 0;
+                    } else {
+                        msgCtx->textColorR = D_801D07DC[(s16)(character - 1)][0] + msgCtx->unk120D4;
+                    }
+
+                    if (D_801D07DC[(s16)(character - 1)][1] + msgCtx->unk120D4 >= 255) {
+                        msgCtx->textColorG = D_801D07DC[(s16)(character - 1)][1];
+                    } else {
+                        msgCtx->textColorG = D_801D07DC[(s16)(character - 1)][1] + msgCtx->unk120D4;
+                    }
+
+                    if (D_801D07DC[(s16)(character - 1)][2] + msgCtx->unk120D4 < 0) {
+                        msgCtx->textColorB = 0;
+                    } else {
+                        msgCtx->textColorB = D_801D07DC[(s16)(character - 1)][2] + msgCtx->unk120D4;
+                    }
+
+                } else if (play->pauseCtx.bombersNotebookOpen) {
+                    msgCtx->textColorR = D_801D089C[(s16)(character - 1)][0];
+                    msgCtx->textColorG = D_801D089C[(s16)(character - 1)][1];
+                    msgCtx->textColorB = D_801D089C[(s16)(character - 1)][2];
+                } else if (msgCtx->textBoxType == TEXTBOX_TYPE_1) {
+                    msgCtx->textColorR = D_801D07DC[(s16)(character - 1)][0];
+                    msgCtx->textColorG = D_801D07DC[(s16)(character - 1)][1];
+                    msgCtx->textColorB = D_801D07DC[(s16)(character - 1)][2];
+                } else if (msgCtx->textBoxType == TEXTBOX_TYPE_D) {
+                    msgCtx->textColorR = D_801D086C[(s16)(character - 1)][0];
+                    msgCtx->textColorG = D_801D086C[(s16)(character - 1)][1];
+                    msgCtx->textColorB = D_801D086C[(s16)(character - 1)][2];
+                } else {
+                    msgCtx->textColorR = D_801D080C[(s16)(character - 1)][0];
+                    msgCtx->textColorG = D_801D080C[(s16)(character - 1)][1];
+                    msgCtx->textColorB = D_801D080C[(s16)(character - 1)][2];
+                }
+
+                if ((i + 1) == msgCtx->textDrawPos) {
+                    msgCtx->textDrawPos++;
+                }
+                break;
+
+            case 0x20: // ` `
+                msgCtx->textPosX += 6;
+                break;
+
+            case 0x11:
+                msgCtx->textPosY += msgCtx->unk11FFC;
+                // fallthrough
```

Comment:
Does MM not have the macro OOT has?

Reply:
macro for `case 0x11` or macro for `FALLTHROUGH`?

Reply:
`FALLTHROUGH`

Reply:
No, MM does not have that macro yet

Reply:
the libultra cleanup PR adds the `FALLTHROUGH` but doesn't use it anywhere yet.
It adds it because I needed some attributes stuff and thought it would be better to bring the whole header over